docs/technical/pull-requests.md

```diff
@@ -0,0 +1,44 @@
+# Checking out pull requests from a forked repository
+PR [#3602](https://github.com/desktop/desktop/pull/3602) introduced the ability to checkout a branch from a forked repository. In order to accomplish this, we needed a way to manage remotes on your behalf. This document is intended to detail the process we developed to make checking out PRs as frictionless as possible.
+
+## Magic Remote Prefix
+One of the main problems we needed to solve was determining which remotes are no longer needed and can be cleaned. We decided to prefix the remotes we add on your behalf with a magic string: `github-desktop-`
```

Comment:
github-desktop-[username]

Reply:
I'm going back and forth with whether we say "github-desktop-" as the prefix or just say that the remote is titled "github-desktop-[username]"... feel free to ignore if you have a strong opinion

Reply:
Could we swap "can be cleaned" with "can be removed"? "Cleaned" may not be clear as it's not git's terminology.